### ISSUE TITLE
sysroot/usr/include/limits.h:125:26: error: no include path in which to search for limits.h

### ISSUE THREAD

**Issue by nephaste** [crosstool-ng/crosstool-ng#47]
Hi 

Thanks for your help on the issue

I am building a ToolChain for Qnap on a debian 6, i used the last ctng from git.
some month ago i successfully built on a Debian 7, using same config, i have error (glibc 2.6.1 + gcc 4.3.6)

[INFO ]  =================================================================
[INFO ]  Installing kernel headers
[EXTRA]    Installing kernel headers
[EXTRA]    Checking installed headers
[INFO ]  Installing kernel headers: done in 8.79s (at 05:23)
[INFO ]  =================================================================
[INFO ]  Installing C library headers & start files
[EXTRA]    Configuring C library
[EXTRA]    Installing C library headers
[EXTRA]    Installing C library start files
[ERROR]    /data/x-tools/i686-Qnap-linux-gnu/i686-Qnap-linux-gnu/sysroot/usr/include/limits.h:125:26: error: no include path in which to search fo                                          r limits.h
[ERROR]    /data/x-tools/i686-Qnap-linux-gnu/i686-Qnap-linux-gnu/sysroot/usr/include/limits.h:125:26: error: no include path in which to search fo                                          r limits.h
[ERROR]    /data/x-tools/i686-Qnap-linux-gnu/i686-Qnap-linux-gnu/sysroot/usr/include/limits.h:125:26: error: no include path in which to search fo                                          r limits.h
[ERROR]    /data/x-tools/i686-Qnap-linux-gnu/i686-Qnap-linux-gnu/sysroot/usr/include/limits.h:125:26: error: no include path in which to search fo                                          r limits.h
[ERROR]    /data/x-tools/i686-Qnap-linux-gnu/i686-Qnap-linux-gnu/sysroot/usr/include/limits.h:125:26: error: no include path in which to search fo                                          r limits.h
[ERROR]    /data/x-tools/i686-Qnap-linux-gnu/i686-Qnap-linux-gnu/sysroot/usr/include/limits.h:125:26: error: no include path in which to search fo                                          r limits.h
[ERROR]    make[3]: *** [/CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/init-first.o] Error 1
[ERROR]    libc-tls.c:36: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'static_dtv'
[ERROR]    libc-tls.c:100: error: 'TLS_TCB_ALIGN' undeclared (first use in this function)
[ERROR]    libc-tls.c:100: error: (Each undeclared identifier is reported only once
[ERROR]    libc-tls.c:100: error: for each function it appears in.)
[ERROR]    libc-tls.c:156:3: error: #error "Either TLS_TCB_AT_TP or TLS_DTV_AT_TP must be defined"
[ERROR]    libc-tls.c:164: error: 'static_dtv' undeclared (first use in this function)
[ERROR]    libc-tls.c:176:3: error: #error "Either TLS_TCB_AT_TP or TLS_DTV_AT_TP must be defined"
[ERROR]    libc-tls.c:193:3: error: #error "Either TLS_TCB_AT_TP or TLS_DTV_AT_TP must be defined"
[ERROR]    libc-tls.c:195: error: 'lossage' undeclared (first use in this function)
[ERROR]    libc-tls.c:220: error: 'TLS_TCB_ALIGN' undeclared (first use in this function)
[ERROR]    libc-tls.c:237: error: 'TLS_TCB_ALIGN' undeclared (first use in this function)
[ERROR]    libc-tls.c:248: error: 'TLS_INIT_TCB_SIZE' undeclared (first use in this function)
[ERROR]    libc-tls.c:248: error: 'TLS_INIT_TCB_ALIGN' undeclared (first use in this function)
[ERROR]    make[3]: *** [/CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/libc-tls.o] Error 1
[ERROR]    make[3]: *** [/CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/libc-start.o] Error 1
[ERROR]    make[2]: *** [csu/subdir_lib] Error 2
[ERROR]    make[1]: *** [csu/subdir_lib] Error 2
[ERROR]
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Installing C library headers & start files'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_DoExecLog[scripts/functions@257]
[ERROR]  >>        called from: do_libc_backend_once[scripts/build/libc/glibc-eglibc.sh-common@438]
[ERROR]  >>        called from: do_libc_backend[scripts/build/libc/glibc-eglibc.sh-common@149]
[ERROR]  >>        called from: do_libc_start_files[scripts/build/libc/glibc-eglibc.sh-common@66]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@638]
[ERROR]  >>
[ERROR]  >>  For more info on this error, look at the file: 'build.log'
[ERROR]  >>  There is a list of known issues, some with workarounds, in:
[ERROR]  >>      '/CROSS/share/doc/crosstool-ng/ct-ng.1.20.0/B - Known issues.txt'
[ERROR]
[ERROR]  (elapsed: 5:45.14)
[05:45] / make: *** [build] Erreur 2

build log report 

[ALL  ]    mv -f /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/version-info.hT /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/version-info.h
[ALL  ]    i686-Qnap-linux-gnu-gcc     ../sysdeps/unix/sysv/linux/init-first.c -c -std=gnu99 -fgnu89-inline -O2 -U_FORTIFY_SOURCE -Wall -Winline -Wwrite-strings -fmerge-all-constants -march=i686 -Wstrict-prototypes -mpreferred-stack-boundary=4     -I../include -I/CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu -I/CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles -I../sysdeps/i386/elf -I../sysdeps/unix/sysv/linux/i386 -I../sysdeps/unix/sysv/linux -I../sysdeps/gnu -I../sysdeps/unix/common -I../sysdeps/unix/mman -I../sysdeps/unix/inet -I../sysdeps/unix/sysv/i386 -I../sysdeps/unix/sysv -I../sysdeps/unix/i386 -I../sysdeps/unix -I../sysdeps/posix -I../sysdeps/i386/i686/fpu -I../sysdeps/i386/i686 -I../sysdeps/i386/i486 -I../sysdeps/i386/fpu -I../sysdeps/i386 -I../sysdeps/wordsize-32 -I../sysdeps/ieee754/ldbl-96 -I../sysdeps/ieee754/dbl-64 -I../sysdeps/ieee754/flt-32 -I../sysdeps/ieee754 -I../sysdeps/generic/elf -I../sysdeps/generic  -I.. -I../libio -I. -nostdinc -isystem /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/buildtools/lib/gcc/i686-Qnap-linux-gnu/4.3.6/include -isystem /data/x-tools/i686-Qnap-linux-gnu/i686-Qnap-linux-gnu/sysroot/usr/include  -include ../include/libc-symbols.h       -DHAVE_INITFINI -o /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/init-first.o -MD -MP -MF /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/init-first.o.dt -MT /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/init-first.o
[ALL  ]    i686-Qnap-linux-gnu-gcc     libc-start.c -c -std=gnu99 -fgnu89-inline -O2 -U_FORTIFY_SOURCE -Wall -Winline -Wwrite-strings -fmerge-all-constants -march=i686 -Wstrict-prototypes -mpreferred-stack-boundary=4     -I../include -I/CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu -I/CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles -I../sysdeps/i386/elf -I../sysdeps/unix/sysv/linux/i386 -I../sysdeps/unix/sysv/linux -I../sysdeps/gnu -I../sysdeps/unix/common -I../sysdeps/unix/mman -I../sysdeps/unix/inet -I../sysdeps/unix/sysv/i386 -I../sysdeps/unix/sysv -I../sysdeps/unix/i386 -I../sysdeps/unix -I../sysdeps/posix -I../sysdeps/i386/i686/fpu -I../sysdeps/i386/i686 -I../sysdeps/i386/i486 -I../sysdeps/i386/fpu -I../sysdeps/i386 -I../sysdeps/wordsize-32 -I../sysdeps/ieee754/ldbl-96 -I../sysdeps/ieee754/dbl-64 -I../sysdeps/ieee754/flt-32 -I../sysdeps/ieee754 -I../sysdeps/generic/elf -I../sysdeps/generic  -I.. -I../libio -I. -nostdinc -isystem /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/buildtools/lib/gcc/i686-Qnap-linux-gnu/4.3.6/include -isystem /data/x-tools/i686-Qnap-linux-gnu/i686-Qnap-linux-gnu/sysroot/usr/include  -include ../include/libc-symbols.h       -DHAVE_INITFINI -o /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/libc-start.o -MD -MP -MF /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/libc-start.o.dt -MT /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/libc-start.o
[ALL  ]    i686-Qnap-linux-gnu-gcc     ../sysdeps/unix/sysv/linux/i386/sysdep.S -c  -I../include -I/CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu -I/CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles -I../sysdeps/i386/elf -I../sysdeps/unix/sysv/linux/i386 -I../sysdeps/unix/sysv/linux -I../sysdeps/gnu -I../sysdeps/unix/common -I../sysdeps/unix/mman -I../sysdeps/unix/inet -I../sysdeps/unix/sysv/i386 -I../sysdeps/unix/sysv -I../sysdeps/unix/i386 -I../sysdeps/unix -I../sysdeps/posix -I../sysdeps/i386/i686/fpu -I../sysdeps/i386/i686 -I../sysdeps/i386/i486 -I../sysdeps/i386/fpu -I../sysdeps/i386 -I../sysdeps/wordsize-32 -I../sysdeps/ieee754/ldbl-96 -I../sysdeps/ieee754/dbl-64 -I../sysdeps/ieee754/flt-32 -I../sysdeps/ieee754 -I../sysdeps/generic/elf -I../sysdeps/generic  -I.. -I../libio -I. -nostdinc -isystem /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/buildtools/lib/gcc/i686-Qnap-linux-gnu/4.3.6/include -isystem /data/x-tools/i686-Qnap-linux-gnu/i686-Qnap-linux-gnu/sysroot/usr/include  -include ../include/libc-symbols.h       -DHAVE_INITFINI -DASSEMBLER  -I/CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/. -DGAS_SYNTAX  -Wa,--noexecstack  -o /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/sysdep.o -MD -MP -MF /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/sysdep.o.dt -MT /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/sysdep.o
[ALL  ]    i686-Qnap-linux-gnu-gcc     version.c -c -std=gnu99 -fgnu89-inline -O2 -U_FORTIFY_SOURCE -Wall -Winline -Wwrite-strings -fmerge-all-constants -march=i686 -Wstrict-prototypes -mpreferred-stack-boundary=4     -I../include -I/CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu -I/CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles -I../sysdeps/i386/elf -I../sysdeps/unix/sysv/linux/i386 -I../sysdeps/unix/sysv/linux -I../sysdeps/gnu -I../sysdeps/unix/common -I../sysdeps/unix/mman -I../sysdeps/unix/inet -I../sysdeps/unix/sysv/i386 -I../sysdeps/unix/sysv -I../sysdeps/unix/i386 -I../sysdeps/unix -I../sysdeps/posix -I../sysdeps/i386/i686/fpu -I../sysdeps/i386/i686 -I../sysdeps/i386/i486 -I../sysdeps/i386/fpu -I../sysdeps/i386 -I../sysdeps/wordsize-32 -I../sysdeps/ieee754/ldbl-96 -I../sysdeps/ieee754/dbl-64 -I../sysdeps/ieee754/flt-32 -I../sysdeps/ieee754 -I../sysdeps/generic/elf -I../sysdeps/generic  -I.. -I../libio -I. -nostdinc -isystem /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/buildtools/lib/gcc/i686-Qnap-linux-gnu/4.3.6/include -isystem /data/x-tools/i686-Qnap-linux-gnu/i686-Qnap-linux-gnu/sysroot/usr/include  -include ../include/libc-symbols.h       -DHAVE_INITFINI -o /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/version.o -MD -MP -MF /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/version.o.dt -MT /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/version.o
[ALL  ]    i686-Qnap-linux-gnu-gcc     check_fds.c -c -std=gnu99 -fgnu89-inline -O2 -U_FORTIFY_SOURCE -Wall -Winline -Wwrite-strings -fmerge-all-constants -march=i686 -Wstrict-prototypes -mpreferred-stack-boundary=4     -I../include -I/CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu -I/CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles -I../sysdeps/i386/elf -I../sysdeps/unix/sysv/linux/i386 -I../sysdeps/unix/sysv/linux -I../sysdeps/gnu -I../sysdeps/unix/common -I../sysdeps/unix/mman -I../sysdeps/unix/inet -I../sysdeps/unix/sysv/i386 -I../sysdeps/unix/sysv -I../sysdeps/unix/i386 -I../sysdeps/unix -I../sysdeps/posix -I../sysdeps/i386/i686/fpu -I../sysdeps/i386/i686 -I../sysdeps/i386/i486 -I../sysdeps/i386/fpu -I../sysdeps/i386 -I../sysdeps/wordsize-32 -I../sysdeps/ieee754/ldbl-96 -I../sysdeps/ieee754/dbl-64 -I../sysdeps/ieee754/flt-32 -I../sysdeps/ieee754 -I../sysdeps/generic/elf -I../sysdeps/generic  -I.. -I../libio -I. -nostdinc -isystem /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/buildtools/lib/gcc/i686-Qnap-linux-gnu/4.3.6/include -isystem /data/x-tools/i686-Qnap-linux-gnu/i686-Qnap-linux-gnu/sysroot/usr/include  -include ../include/libc-symbols.h       -DHAVE_INITFINI -o /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/check_fds.o -MD -MP -MF /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/check_fds.o.dt -MT /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/check_fds.o
[ALL  ]    i686-Qnap-linux-gnu-gcc     libc-tls.c -c -std=gnu99 -fgnu89-inline -O2 -U_FORTIFY_SOURCE -Wall -Winline -Wwrite-strings -fmerge-all-constants -march=i686 -Wstrict-prototypes -mpreferred-stack-boundary=4     -I../include -I/CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu -I/CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles -I../sysdeps/i386/elf -I../sysdeps/unix/sysv/linux/i386 -I../sysdeps/unix/sysv/linux -I../sysdeps/gnu -I../sysdeps/unix/common -I../sysdeps/unix/mman -I../sysdeps/unix/inet -I../sysdeps/unix/sysv/i386 -I../sysdeps/unix/sysv -I../sysdeps/unix/i386 -I../sysdeps/unix -I../sysdeps/posix -I../sysdeps/i386/i686/fpu -I../sysdeps/i386/i686 -I../sysdeps/i386/i486 -I../sysdeps/i386/fpu -I../sysdeps/i386 -I../sysdeps/wordsize-32 -I../sysdeps/ieee754/ldbl-96 -I../sysdeps/ieee754/dbl-64 -I../sysdeps/ieee754/flt-32 -I../sysdeps/ieee754 -I../sysdeps/generic/elf -I../sysdeps/generic  -I.. -I../libio -I. -nostdinc -isystem /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/buildtools/lib/gcc/i686-Qnap-linux-gnu/4.3.6/include -isystem /data/x-tools/i686-Qnap-linux-gnu/i686-Qnap-linux-gnu/sysroot/usr/include  -include ../include/libc-symbols.h       -DHAVE_INITFINI -o /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/libc-tls.o -MD -MP -MF /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/libc-tls.o.dt -MT /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/libc-tls.o
[ALL  ]    i686-Qnap-linux-gnu-gcc     elf-init.c -c -std=gnu99 -fgnu89-inline -O2 -U_FORTIFY_SOURCE -Wall -Winline -Wwrite-strings -fmerge-all-constants -march=i686 -Wstrict-prototypes -mpreferred-stack-boundary=4     -I../include -I/CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu -I/CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles -I../sysdeps/i386/elf -I../sysdeps/unix/sysv/linux/i386 -I../sysdeps/unix/sysv/linux -I../sysdeps/gnu -I../sysdeps/unix/common -I../sysdeps/unix/mman -I../sysdeps/unix/inet -I../sysdeps/unix/sysv/i386 -I../sysdeps/unix/sysv -I../sysdeps/unix/i386 -I../sysdeps/unix -I../sysdeps/posix -I../sysdeps/i386/i686/fpu -I../sysdeps/i386/i686 -I../sysdeps/i386/i486 -I../sysdeps/i386/fpu -I../sysdeps/i386 -I../sysdeps/wordsize-32 -I../sysdeps/ieee754/ldbl-96 -I../sysdeps/ieee754/dbl-64 -I../sysdeps/ieee754/flt-32 -I../sysdeps/ieee754 -I../sysdeps/generic/elf -I../sysdeps/generic  -I.. -I../libio -I. -nostdinc -isystem /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/buildtools/lib/gcc/i686-Qnap-linux-gnu/4.3.6/include -isystem /data/x-tools/i686-Qnap-linux-gnu/i686-Qnap-linux-gnu/sysroot/usr/include  -include ../include/libc-symbols.h       -DHAVE_INITFINI -o /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/elf-init.o -MD -MP -MF /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/elf-init.o.dt -MT /CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/elf-init.o
[ALL  ]    In file included from ../include/limits.h:125,
[ALL  ]                     from ../sysdeps/unix/sysv/linux/sys/param.h:22,
[ALL  ]                     from ../sysdeps/unix/sysv/linux/init-first.c:26:
[ERROR]    /data/x-tools/i686-Qnap-linux-gnu/i686-Qnap-linux-gnu/sysroot/usr/include/limits.h:125:26: error: no include path in which to search for limits.h
[ALL  ]    In file included from ../include/limits.h:125,
[ALL  ]                     from ../sysdeps/unix/sysv/linux/sys/param.h:22,
[ALL  ]                     from ../sysdeps/i386/i686/hp-timing.h:25,
[ALL  ]                     from ../sysdeps/generic/ldsodefs.h:39,
[ALL  ]                     from ../sysdeps/unix/sysv/linux/ldsodefs.h:25,
[ALL  ]                     from libc-start.c:22:
[ERROR]    /data/x-tools/i686-Qnap-linux-gnu/i686-Qnap-linux-gnu/sysroot/usr/include/limits.h:125:26: error: no include path in which to search for limits.h
[ALL  ]    In file included from ../include/limits.h:125,
[ALL  ]                     from ../sysdeps/unix/sysv/linux/sys/param.h:22,
[ALL  ]                     from ../sysdeps/i386/i686/hp-timing.h:25,
[ALL  ]                     from ../sysdeps/generic/ldsodefs.h:39,
[ALL  ]                     from ../sysdeps/unix/sysv/linux/ldsodefs.h:25,
[ALL  ]                     from libc-tls.c:21:
[ERROR]    /data/x-tools/i686-Qnap-linux-gnu/i686-Qnap-linux-gnu/sysroot/usr/include/limits.h:125:26: error: no include path in which to search for limits.h
[ALL  ]    In file included from ../include/limits.h:125,
[ALL  ]                     from ../stdio-common/_itoa.h:23,
[ALL  ]                     from ../sysdeps/i386/i686/hp-timing.h:26,
[ALL  ]                     from ../sysdeps/generic/ldsodefs.h:39,
[ALL  ]                     from ../sysdeps/unix/sysv/linux/ldsodefs.h:25,
[ALL  ]                     from libc-start.c:22:
[ERROR]    /data/x-tools/i686-Qnap-linux-gnu/i686-Qnap-linux-gnu/sysroot/usr/include/limits.h:125:26: error: no include path in which to search for limits.h
[ALL  ]    In file included from ../include/limits.h:125In file included from ../include/limits.h:125,
[ALL  ]                     from ../stdio-common/_itoa.h:23,
[ALL  ]                     from ../sysdeps/i386/i686/hp-timing.h:26,
[ALL  ]                     from ../include/libc-internal.h:7,
[ALL  ]                     from ../sysdeps/unix/sysv/linux/init-first.c:28:
[ERROR]    /data/x-tools/i686-Qnap-linux-gnu/i686-Qnap-linux-gnu/sysroot/usr/include/limits.h:125:26: error: no include path in which to search for limits.h
[ALL  ]    ,
[ALL  ]                     from ../stdio-common/_itoa.h:23,
[ALL  ]                     from ../sysdeps/i386/i686/hp-timing.h:26,
[ALL  ]                     from ../sysdeps/generic/ldsodefs.h:39,
[ALL  ]                     from ../sysdeps/unix/sysv/linux/ldsodefs.h:25,
[ALL  ]                     from libc-tls.c:21:
[ERROR]    /data/x-tools/i686-Qnap-linux-gnu/i686-Qnap-linux-gnu/sysroot/usr/include/limits.h:125:26: error: no include path in which to search for limits.h
[ERROR]    make[3]: *** [/CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/init-first.o] Error 1
[ALL  ]    make[3]: *** Waiting for unfinished jobs....
[ERROR]    libc-tls.c:36: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'static_dtv'
[ALL  ]    libc-tls.c: In function 'init_static_tls':
[ERROR]    libc-tls.c:100: error: 'TLS_TCB_ALIGN' undeclared (first use in this function)
[ERROR]    libc-tls.c:100: error: (Each undeclared identifier is reported only once
[ERROR]    libc-tls.c:100: error: for each function it appears in.)
[ERROR]    libc-tls.c:156:3: error: #error "Either TLS_TCB_AT_TP or TLS_DTV_AT_TP must be defined"
[ALL  ]    libc-tls.c: In function '__libc_setup_tls':
[ERROR]    libc-tls.c:164: error: 'static_dtv' undeclared (first use in this function)
[ERROR]    libc-tls.c:176:3: error: #error "Either TLS_TCB_AT_TP or TLS_DTV_AT_TP must be defined"
[ERROR]    libc-tls.c:193:3: error: #error "Either TLS_TCB_AT_TP or TLS_DTV_AT_TP must be defined"
[ERROR]    libc-tls.c:195: error: 'lossage' undeclared (first use in this function)
[ERROR]    libc-tls.c:220: error: 'TLS_TCB_ALIGN' undeclared (first use in this function)
[ALL  ]    libc-tls.c:118: warning: unused variable 'tcb_offset'
[ALL  ]    libc-tls.c: In function '_dl_tls_setup':
[ERROR]    libc-tls.c:237: error: 'TLS_TCB_ALIGN' undeclared (first use in this function)
[ALL  ]    libc-tls.c: In function '__pthread_initialize_minimal':
[ERROR]    libc-tls.c:248: error: 'TLS_INIT_TCB_SIZE' undeclared (first use in this function)
[ERROR]    libc-tls.c:248: error: 'TLS_INIT_TCB_ALIGN' undeclared (first use in this function)
[ERROR]    make[3]: *** [/CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/libc-tls.o] Error 1
[ERROR]    make[3]: *** [/CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles/csu/libc-start.o] Error 1
[ALL  ]    make[3]: Leaving directory `/CROSS/glibc-2.6.1/csu'
[ERROR]    make[2]: *** [csu/subdir_lib] Error 2
[ALL  ]    make[2]: Leaving directory `/CROSS/glibc-2.6.1'
[ERROR]    make[1]: *** [csu/subdir_lib] Error 2
[ALL  ]    make[1]: Leaving directory `/CROSS/crosstool-ng/.build/i686-Qnap-linux-gnu/build/build-libc-startfiles'
[ERROR]  
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Installing C library headers & start files'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_DoExecLog[scripts/functions@257]
[ERROR]  >>        called from: do_libc_backend_once[scripts/build/libc/glibc-eglibc.sh-common@438]
[ERROR]  >>        called from: do_libc_backend[scripts/build/libc/glibc-eglibc.sh-common@149]
[ERROR]  >>        called from: do_libc_start_files[scripts/build/libc/glibc-eglibc.sh-common@66]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@638]
[ERROR]  >>
[ERROR]  >>  For more info on this error, look at the file: 'build.log'
[ERROR]  >>  There is a list of known issues, some with workarounds, in:
[ERROR]  >>      '/CROSS/share/doc/crosstool-ng/ct-ng.1.20.0/B - Known issues.txt'
[ERROR]  
[ERROR]  (elapsed: 5:45.14)

thanks a lot for your help,

**Comment by bhundven** [comment]
Please post a gist to your .config and full build log.

Thanks,

-Bryan

**Comment by nephaste** [comment]
Hi,
I reproduced it on a Fedora 21 x86 (same issue)
find logs http://www.positiv-it.fr/ctng_logs.tgz

look like an header conflict to me...

I finally used CroosTool-NG version 1.16 and built a toolchain succesfully, but i would prefer use la ct-ng version.

thanks for your feedback,

**Comment by bhundven** [comment]
Does that have the .config in it?

**Comment by nephaste** [comment]
yes .config and build.log

**Comment by bhundven** [comment]
Great. I'll check it out. Thanks!

**Comment by bhundven** [comment]
```CT_LIBC_GLIBC_CUSTOM_LOCATION="/CROSS/glibc-2.6.1"```
Yea... so, I will validate that glibc-2.6.1 *will not work* with newer versions of crosstool-ng.

Every version of crosstool-ng has a branch where updates can happen. While I'm currently maintaining the whole tree, my focus is on the master branch. I'm happy to accept patches for older branches, if you need some change backported to 1.16.x, just do a pull request like normal and I'll be happy to review/test/apply.

**Comment by bhundven** [comment]
For now I'm going to close this issue with 'wontfix'.
